app/src/main/java/de/danoeh/antennapod/activity/VideoplayerActivity.java

```diff
@@ -480,4 +484,102 @@ void stop() {
 
     }
 
+
+    //Hardware keyboard support
+    @Override
+    public boolean onKeyUp(int keyCode, KeyEvent event) {
+        AudioManager audioManager = (AudioManager) getSystemService(AUDIO_SERVICE);
+        Float videoPercent = null;
+
+        switch (keyCode) {
+            case KeyEvent.KEYCODE_ENTER: //Fallthrough
+            case KeyEvent.KEYCODE_SPACE:
+                //Play/Pause
+                onPlayPause();
+                toggleVideoControlsVisibility();
+                break;
+            case KeyEvent.KEYCODE_DPAD_LEFT:
+                //Go Back
+                onRewind();
+                showSkipAnimation(false);
+                break;
+            case KeyEvent.KEYCODE_DPAD_RIGHT:
+                //Go Forward
+                onFastForward();
+                showSkipAnimation(true);
+                break;
+            case KeyEvent.KEYCODE_M:
+                //Mute/Unmute
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    audioManager.adjustStreamVolume(AudioManager.STREAM_MUSIC, AudioManager.ADJUST_TOGGLE_MUTE, 0);
+                } else {
+                    if (audioManager.getStreamVolume(AudioManager.STREAM_MUSIC) == 0) {
+                        //Unmute
+                        audioManager.setStreamVolume(AudioManager.STREAM_MUSIC,
+                                audioManager.getStreamMaxVolume(AudioManager.STREAM_MUSIC), 0);
+                    } else {
+                        //Mute
+                        audioManager.setStreamVolume(AudioManager.STREAM_MUSIC, 0, 0);
+                    }
+                }
+                break;
+            case KeyEvent.KEYCODE_DPAD_UP:
+                //Raise volume
+                audioManager.adjustStreamVolume(AudioManager.STREAM_MUSIC,
+                        AudioManager.ADJUST_RAISE, AudioManager.FLAG_SHOW_UI);
+                break;
+            case KeyEvent.KEYCODE_DPAD_DOWN:
+                //Raise volume
+                audioManager.adjustStreamVolume(AudioManager.STREAM_MUSIC,
+                        AudioManager.ADJUST_LOWER, AudioManager.FLAG_SHOW_UI);
+                break;
+            case KeyEvent.KEYCODE_ESCAPE:
+            case KeyEvent.KEYCODE_F:
+                //Exit fullscreen mode
+                onOptionsItemSelected(new ActionMenuItem(this, 0, android.R.id.home, 0, 0, ""));
```

Comment:
This looks a bit strange. Doesn't Android have key bindings for going back, anyway?